test/api_integration/apis/saved_objects/export.js

```diff
@@ -0,0 +1,363 @@
+                validation: { source: 'payload', keys: [ 'objects.0.type' ] },
+              });
+            });
+        });
+      });
+
+      describe('10,000 objects', () => {
+        before(() => esArchiver.load('saved_objects/10k'));
+        after(() => esArchiver.unload('saved_objects/10k'));
+
+        it('should return 400 when exporting without type or objects passed in', async () => {
+          await supertest
+            .post('/api/saved_objects/_export')
+            .expect(400)
+            .then((resp) => {
+              expect(resp.body).to.eql({
+                statusCode: 400,
+                error: 'Bad Request',
+                message: '"value" must be an object',
+                validation: { source: 'payload', keys: [ 'value' ] },
+              });
+            });
+        });
+
+        it('should return 200 when exporting by single type', async () => {
+          await supertest
+            .post('/api/saved_objects/_export')
+            .send({
+              type: 'dashboard',
+            })
+            .expect(200)
+            .then((resp) => {
+              expect(resp.headers['content-disposition']).to.eql('attachment; filename="export.ndjson"');
+              expect(resp.headers['content-type']).to.eql('application/ndjson');
+              const objects = resp.text.split('\n').map(JSON.parse);
+              expect(objects).to.eql([{
+                attributes: {
+                  description: '',
+                  hits: 0,
+                  kibanaSavedObjectMeta: {
+                    searchSourceJSON: objects[0].attributes.kibanaSavedObjectMeta.searchSourceJSON,
+                  },
+                  optionsJSON: objects[0].attributes.optionsJSON,
+                  panelsJSON: objects[0].attributes.panelsJSON,
+                  refreshInterval: {
+                    display: 'Off',
+                    pause: false,
+                    value: 0,
+                  },
+                  timeFrom: 'Wed Sep 16 2015 22:52:17 GMT-0700',
+                  timeRestore: true,
+                  timeTo: 'Fri Sep 18 2015 12:24:38 GMT-0700',
+                  title: 'Requests',
+                  uiStateJSON: '{}',
+                  version: 1,
+                },
+                id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+                migrationVersion: {
+                  dashboard: '7.0.0',
+                },
+                references: [
+                  {
+                    id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+                    name: 'panel_0',
+                    type: 'visualization',
+                  },
+                ],
+                type: 'dashboard',
+                updated_at: '2017-09-21T18:57:40.826Z',
+                version: objects[0].version,
+              }]);
+              JSON.parse(objects[0].attributes.kibanaSavedObjectMeta.searchSourceJSON);
```

Comment:
I assume this and the lines below are asserting that these are valid JSON, but that'd be a little more clear if we wrapped them in `expect(() => ...).not.to.throwError()`